Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1149,6 +1149,12 @@ static String mutationObserverNotificationString()
         m_activeAnnotation->updateGeometry();
 #endif
 
+#if ENABLE(UNIFIED_PDF_DATA_DETECTION)
+    auto lastKnownMousePositionInDocumentSpace = convertDown<FloatPoint>(CoordinateSpace::Plugin, CoordinateSpace::PDFDocumentLayout, lastKnownMousePositionInView());
+    auto pageIndex = pageIndexForDocumentPoint(lastKnownMousePositionInDocumentSpace);
+    dataDetectorOverlayController().didInvalidateHighlightOverlayRects(pageIndex);
+#endif
```

Comment:
Seems like this could be in its own method, as it's duplicated code?

Reply:
Yes that's a good point. I'll make it a separate function before landing.

Reply:
Fixed in 16dbd8a